### PR TITLE
Correcting variational distance calculation

### PR DESCRIPTION
**Pull request by RiccardoRossi**
@pablobecker this PR replaces the use of penalty+enrichment by the use of a geometric based criterion in the calculation of the distances. I tried it in a small 3D test, however it does require much more validation

### CODE REVIEW

File: kratos/elements/levelset_convection_element_simplex.h [hunk 1]
```diff
@@ -189,7 +189,7 @@ class LevelSetConvectionElementSimplex
             const double norm_vel = norm_2(vel_gauss);
             array_1d<double, TNumNodes > a_dot_grad = prod(DN_DX, vel_gauss);
 
-            const double tau_denom = std::max(dyn_st_beta *dt_inv + 2.0 * norm_vel / h + beta*div_v,  1e-2);
+            const double tau_denom = std::max(dyn_st_beta *dt_inv + 2.0 * norm_vel / h + std::abs(beta*div_v),  1e-2);
```
Comment (jcotela) [inline kratos/elements/levelset_convection_element_simplex.h]: Note that beta is hardcoded to be 0 in line 165... I don't see the use of this change.

File: kratos/processes/variational_distance_calculation_process.h [hunk 2]
```diff
@@ -213,17 +213,92 @@ class VariationalDistanceCalculationProcess
...
+                for(unsigned int i=0; i<TDim+1; i++)
+                {
+                    
+                    double& d = geom[i].FastGetSolutionStepValue(DISTANCE);
+                    if(std::abs(d) > std::abs(distances[i])) d = distances[i];
+                    geom[i].Fix(DISTANCE);
+                }
+            }
+        }
+        mp_distance_model_part->GetCommunicator().SynchronizeCurrentDataToMin(DISTANCE);
```
Comment (jcotela) [inline kratos/processes/variational_distance_calculation_process.h]: If distance is signed, this is a problem. (However, I don't fully understand if distance is signed here)

File: kratos/processes/variational_distance_calculation_process.h [hunk 3]
```diff
@@ -213,17 +213,112 @@ class VariationalDistanceCalculationProcess
...
+                distances[i] = geom[i].GetValue(DISTANCE);
+            
+            array_1d<double,TDim+1> original_distances = distances;
+            
+            unsigned int positives = 0, negatives=0;
+            for(unsigned int i=0; i<TDim+1; i++)
+            {
+                if(distances[i] >= 0) positives++;
+                else negatives++;
+            }
```
Comment (pooyan-dadvand) [inline kratos/processes/variational_distance_calculation_process.h:262]: I think that we have these loop in many places....

**Review by pooyan-dadvand — APPROVED** [review]
:+1: 

My only comment is the duplication of the code. But I'm not sure how to avoid it....

**Comment by RiccardoRossi** [comment]
I am not sure either.

we could have a 

       distance_utilities::IsSplit(distances)

but i am convinced you also don't love this. I will merge the PR but let's think to a clean solution

**Pull request merged**